great_expectations/checkpoint/actions.py

```diff
@@ -234,20 +236,32 @@ def _validate_renderer(cls, renderer: dict | SlackRenderer) -> SlackRenderer:
 
     @root_validator
     def _root_validate_slack_params(cls, values: dict) -> dict:
-        slack_token = values["slack_token"]
-        slack_channel = values["slack_channel"]
-        slack_webhook = values["slack_webhook"]
+        from great_expectations.data_context.data_context.context_factory import project_manager
+
+        config_provider = project_manager.get_config_provider()
+        for credential in ("slack_webhook", "slack_token", "slack_channel"):
+            if isinstance(values[credential], ConfigStr):
+                values[credential] = cls._substitute_slack_credential(
+                    slack_credential=values[credential],
+                    config_provider=config_provider,
+                )
```

Comment:
@cdkini sorry I missed this in the review.
The values shouldn't be updated in the validator method. They should only be substituted when they are actually needed which looks like that would be inside the `.run()` method.

- https://github.com/great-expectations/great_expectations/issues/10473